config/vm_repo.go

```diff
@@ -63,3 +67,61 @@ func (r vMRepo) ClearCurrent() error {
 	}
 	return nil
 }
+
+func (r vMRepo) FindCurrentIP() (string, bool, error) {
+	deploymentState, err := r.deploymentStateService.Load()
+	if err != nil {
+		return "", false, bosherr.WrapError(err, "Loading existing config")
+	}
+
+	currentIP := deploymentState.CurrentIP
+	if currentIP != "" {
+		return currentIP, true, nil
+	}
+
+	return "", false, nil
+}
+
+func (r vMRepo) UpdateCurrentIP(ip string) error {
+	deploymentState, err := r.deploymentStateService.Load()
+	if err != nil {
+		return bosherr.WrapError(err, "Loading existing config")
+	}
+
+	deploymentState.CurrentIP = ip
+
+	err = r.deploymentStateService.Save(deploymentState)
+	if err != nil {
+		return bosherr.WrapError(err, "Saving new config")
+	}
+	return nil
+}
+
+func (r vMRepo) FindCurrentAgentId() (string, bool, error) {
+	deploymentState, err := r.deploymentStateService.Load()
+	if err != nil {
+		return "", false, bosherr.WrapError(err, "Loading existing config")
+	}
+
+	currentAgentID := deploymentState.CurrentAgentID
```

Comment:
Removed current agent ID related, as there's no need to reuse the current agent ID even we are os-reloading the vm. So every time it will generate a new agent ID.